### PR TITLE
[AIDAPP-323]: Enhance the display of articles by all articles, featured articles, and most viewed articles

### PR DESCRIPTION
**Pull request by amit-canyon**
### Ticket(s) or GitHub Issue

- https://canyongbs.atlassian.net/browse/AIDAPP-323

### Technical Description

> Enhance the display of articles by all articles, featured articles, and most viewed articles

### Any deployment steps required?

> No

### Are any Feature Flags and/or Data Migrations that can eventually be removed Added?

> No

_______________________________________________

### CODE REVIEW

File: app-modules/portal/src/Http/Controllers/KnowledgeManagementPortal/KnowledgeManagementPortalCategoryController.php [hunk 1]
```diff
@@ -74,6 +74,12 @@ public function show(KnowledgeBaseCategory $category): JsonResponse
             'articles' => $category->knowledgeBaseItems()
                 ->with('tags')
                 ->public()
+                ->when(request()->get('filter') === 'featured', function ($q) {
+                    $q->where('is_featured', true);
+                })
+                ->when(request()->get('filter') === 'most-viewed', function ($q) {
+                    $q->orderBy('portal_view_count', 'desc');
+                })
```
Comment (Orrison) [inline app-modules/portal/src/Http/Controllers/KnowledgeManagementPortal/KnowledgeManagementPortalCategoryController.php:82]: For both of these functions lets change the variable name to `$query` and also typehint `Builder`

File: app-modules/portal/src/Http/Controllers/KnowledgeManagementPortal/KnowledgeManagementPortalSearchController.php [hunk 2]
```diff
@@ -69,6 +69,12 @@ public function get(Request $request): KnowledgeManagementSearchData
                 ->with('tags')
                 ->when($search->isNotEmpty(), fn (Builder $query) => $query->tap(new SearchBy('title', $search)))
                 ->when($tags->isNotEmpty(), fn (Builder $query) => $query->whereHas('tags', fn (Builder $query) => $query->whereIn('id', $tags)))
+                ->when($request->get('filter') === 'featured', function ($q) {
+                    $q->where('is_featured', true);
+                })
+                ->when($request->get('filter') === 'most-viewed', function ($q) {
+                    $q->orderBy('portal_view_count', 'desc');
+                })
```
Comment (Orrison) [inline app-modules/portal/src/Http/Controllers/KnowledgeManagementPortal/KnowledgeManagementPortalSearchController.php:77]: For both of these functions lets change the variable name to `$query` and also typehint `Builder`

File: app-modules/portal/src/Http/Controllers/KnowledgeManagementPortal/KnowledgeManagementPortalCategoryController.php [hunk 3]
```diff
@@ -74,6 +74,12 @@ public function show(KnowledgeBaseCategory $category): JsonResponse
             'articles' => $category->knowledgeBaseItems()
                 ->with('tags')
                 ->public()
+                ->when(request()->get('filter') === 'featured', function ($q) {
+                    $q->where('is_featured', true);
+                })
+                ->when(request()->get('filter') === 'most-viewed', function ($q) {
+                    $q->orderBy('portal_view_count', 'desc');
```
Comment (Orrison) [inline app-modules/portal/src/Http/Controllers/KnowledgeManagementPortal/KnowledgeManagementPortalCategoryController.php:81]: This should also filter to only show those that have been viewed at all.

File: app-modules/portal/src/Http/Controllers/KnowledgeManagementPortal/KnowledgeManagementPortalSearchController.php [hunk 4]
```diff
@@ -69,6 +69,12 @@ public function get(Request $request): KnowledgeManagementSearchData
                 ->with('tags')
                 ->when($search->isNotEmpty(), fn (Builder $query) => $query->tap(new SearchBy('title', $search)))
                 ->when($tags->isNotEmpty(), fn (Builder $query) => $query->whereHas('tags', fn (Builder $query) => $query->whereIn('id', $tags)))
+                ->when($request->get('filter') === 'featured', function ($q) {
+                    $q->where('is_featured', true);
+                })
+                ->when($request->get('filter') === 'most-viewed', function ($q) {
+                    $q->orderBy('portal_view_count', 'desc');
```
Comment (Orrison) [inline app-modules/portal/src/Http/Controllers/KnowledgeManagementPortal/KnowledgeManagementPortalSearchController.php:76]: This should also filter to only show those that have been viewed at all.

File: portals/knowledge-management/src/Components/FilterComponent.vue [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        <li
+            role="presentation"
+            @click="updateFilter('all-articles')"
+            class="text-sm font-semibold text-primary-600 px-4 py-3 hover:border-primary-600 cursor-pointer"
+            :class="
+                selectedFilter === '' || selectedFilter === 'all-articles'
+                    ? 'border-b-2 border-primary-600'
+                    : 'border-transparent'
+            "
+            data-twe-toggle="pill"
```
Comment (Orrison) [inline portals/knowledge-management/src/Components/FilterComponent.vue:62]: What is the purpose of this `data-twe-toggle` attribute?

File: portals/knowledge-management/src/Components/FilterComponent.vue [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+            aria-selected="true"
+        >
+            All Articles
+        </li>
+        <li
+            role="presentation"
+            @click="updateFilter('featured')"
+            class="text-sm font-semibold text-primary-600 px-4 py-3 border-b-2 hover:border-primary-600 cursor-pointer"
+            :class="selectedFilter === 'featured' ? 'border-b-2 border-primary-600' : 'border-transparent'"
+            data-twe-toggle="pill"
```
Comment (Orrison) [inline portals/knowledge-management/src/Components/FilterComponent.vue:72]: What is the purpose of this `data-twe-toggle` attribute?

File: portals/knowledge-management/src/Components/FilterComponent.vue [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+            aria-selected="true"
+        >
+            Featured
+        </li>
+        <li
+            role="presentation"
+            @click="updateFilter('most-viewed')"
+            class="text-sm font-semibold text-primary-600 px-4 py-3 border-b-2 hover:border-primary-600 cursor-pointer"
+            :class="selectedFilter === 'most-viewed' ? 'border-b-2 border-primary-600' : 'border-transparent'"
+            data-twe-toggle="pill"
```
Comment (Orrison) [inline portals/knowledge-management/src/Components/FilterComponent.vue:82]: What is the purpose of this `data-twe-toggle` attribute?

**Review by Orrison — CHANGES_REQUESTED** [review]

File: portals/knowledge-management/src/Components/FilterComponent.vue (hunk 5 above)
Comment (amit-canyon) [reply portals/knowledge-management/src/Components/FilterComponent.vue:62]: Not now.

**Review by Orrison — APPROVED** [review]

**Pull request merged**